canopeum_backend/canopeum_backend/serializers.py

```diff
@@ -580,6 +590,58 @@ def get_sponsors(self, obj) -> list[str]:
         return [batch.sponsor for batch in batches if batch.sponsor]
 
 
+class SiteDetailSummarySerializer(serializers.ModelSerializer[Site]):
+    site_type = SiteTypeSerializer()
+    coordinate = CoordinatesSerializer()
+    plant_count = serializers.SerializerMethodField()
+    survived_count = serializers.SerializerMethodField()
+    propagation_count = serializers.SerializerMethodField()
+    progress = serializers.SerializerMethodField()
+    sponsors = serializers.SerializerMethodField()
+    admins = SiteAdminSerializer(source="siteadmin_set", many=True)
+    batches = BatchDetailSerializer(source="batch_set", read_only=True, many=True)
+    weather = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Site
+        fields = (
+            "id",
+            "name",
+            "coordinate",
+            "site_type",
+            "plant_count",
+            "survived_count",
+            "propagation_count",
+            "visitor_count",
+            "sponsors",
+            "progress",
+            "admins",
+            "batches",
+            "weather",
+        )
```

Comment:
If you need all fields, consider trying (dunno if this works for your use case, but if it does, then that's nice)
```suggestion
        fields = "__all__"
```

Reply:
We need to let it written like this to not retrieve fields in model which we do not need